Jenkinsfile

```diff
@@ -0,0 +1,25 @@
+node {
+    dir("/root/"){
+    checkout scm
+
+    env.DOCKER_API_VERSION="1.23"
+    appName = "default/flask-app"
+    registryHost = "mycluster.icp:8500/"
+    imageName = "${registryHost}${appName}:${env.BUILD_ID}"
+    env.BUILDIMG=imageName
+    docker.withRegistry('https://mycluster.icp:8500/', 'docker'){
+    stage "Build"
+
+        def pcImg = docker.build("mycluster.icp:8500/default/flask-app:${env.BUILD_ID}", "-f Dockerfile.ppc64le .")
```

Comment:
Instead of hardcoding the registry server, can you set the values in env variable and reference it here.
eg.
env.REGISTRY=mycluster.icp:8500
env.REGISTRY_NAMESPACE=default